src/core/components/info.jsx

```diff
@@ -100,17 +101,20 @@ export default class Info extends React.Component {
     basePath: PropTypes.string,
     externalDocs: ImPropTypes.map,
     getComponent: PropTypes.func.isRequired,
+    oas3selectors: PropTypes.func,
+    selectedServer: PropTypes.string,
   }
 
   render() {
-    let { info, url, host, basePath, getComponent, externalDocs } = this.props
+    let { info, url, host, basePath, getComponent, externalDocs, selectedServer } = this.props
     let version = info.get("version")
     let description = info.get("description")
     let title = info.get("title")
-    let termsOfService = info.get("termsOfService")
+    let termsOfService = buildUrl(info.get("termsOfService"), selectedServer)
     let contact = info.get("contact")
     let license = info.get("license")
-    const { url:externalDocsUrl, description:externalDocsDescription } = (externalDocs || fromJS({})).toJS()
+    let { url:externalDocsUrl, description:externalDocsDescription } = (externalDocs || fromJS({})).toJS()
+    externalDocsUrl = buildUrl( externalDocs.url, selectedServer )
```

Comment:
**Change request**: would prefer to use `ImmutableMap.getIn` to set `externalDocsUrl` and `externalDocsDescription`, instead of futzing with `toJS`. Again, aware that this PR didn't introduce it, but since it's been touched let's improve it!

Would also like to avoid reassignment on `externalDocsUrl` - see my note on operation-tag.jsx L55, same idea here.